efficient_full_table_scan.go

```diff
@@ -0,0 +1,200 @@
+// THIS SCRIPT ASSUMES USING A SINGLE CLIENT TO RUN THE FULL TABLE SCAN
+// Prerequisite:
+//  Install go 1.7: https://www.digitalocean.com/community/tutorials/how-to-install-go-1-7-on-centos-7
+//  Install gocql driver: https://academy.datastax.com/resources/getting-started-apache-cassandra-and-go
+
+package main
+
+import (
+        "fmt"
+        "math"
+        "math/rand"
+        "os"
+        "sync"
+        "sync/atomic"
+        "time"
+
+        "github.com/gocql/gocql"
+)
+
+// Your Scylla cluster setup
+const (
+        // NodesInCluster - the number of nodes in your cluster
+        NodesInCluster = 3
+        // CoresInNode - the number of cores in each node
+        CoresInNode = 8
+        // SmudgeFactor - is a factor to add to make parallelism cooler
+        SmudgeFactor = 3
+        // NumberOfParallelClientThreads - the calculated number of parallel threads the client should run
+        NumberOfParallelClientThreads = NodesInCluster * CoresInNode * SmudgeFactor
+)
+
+// Modify the query for the relevant keyspace.table_name you wish to scan
+// We recommend selecting only required columns, as this can eliminate unnecessary network traffic and processing
+var (
+        queryTemplate = "SELECT token(key) FROM keyspace1.standard1 WHERE token(key) >= %d AND token(key) <= %d;"
+)
+
+type tokenRange struct {
+        StartRange int64
+        EndRange   int64
+}
+
+// Calculates the token range values to be executed in parallel
+func getTokenRanges() []*tokenRange {
+        var n = NumberOfParallelClientThreads
+        var m = int64(n * 100)
+        var maxSize uint64 = math.MaxInt64 * 2
+        var rangeSize = maxSize / uint64(m)
+
+        var start int64 = math.MinInt64
+        var end int64
+        var shouldBreak = false
+
+        var ranges = make([]*tokenRange, m)
+
+        for i := int64(0); i < m; i++ {
+                end = start + int64(rangeSize)
+                if start > 0 && end < 0 {
+                        end = math.MaxInt64
+                        shouldBreak = true
+                }
+
+                ranges[i] = &tokenRange{StartRange: start, EndRange: end}
+
+                if shouldBreak {
+                        break
+                }
+
+                start = end + 1
+        }
+
+        return ranges
+}
+
+// Randomize the execution of the token range statements
+func shuffle(data []*tokenRange) {
+        for i := 1; i < len(data); i++ {
+                r := rand.Intn(i + 1)
+                if i != r {
+                        data[r], data[i] = data[i], data[r]
+                }
+        }
+}
+
+func main() {
+        var totalRows uint64
+        var ranges = getTokenRanges()
+
+        shuffle(ranges)
+		
+		//Cluster IPs and other variables
+        cluster := gocql.NewCluster("10.240.0.29", "10.240.0.30", "10.240.0.35")
```

Comment:
reading Scylla IPs from the command line will make it easier to run the example without compiling it